wiring/src/spark_wiring_ble.cpp

```diff
@@ -2798,7 +2799,9 @@ int BleLocalDevice::disconnect(const BlePeerDevice& peer) const {
 
 int BleLocalDevice::disconnectAll() const {
     WiringBleLock lk;
-    for (auto& p : impl()->peers()) {
+    // DO NOT use auto&, otherwise, any operation using BlePeerDevice::impl() after the peer device
+    // being removed from the peers() vector may not be guaranteed. See BlePeerDevice::disconnect().
+    for (auto p : impl()->peers()) {
```

Comment:
How does this work? `removeOne` will not work because you are working on a copy here. Removing while iterating usually can be implemented with a regular for loop instead of a range-for.

Reply:
The `removeOne()` will compare with a const reference and remove the one that is `==` with the reference, so the parameter doesn't have to be exactly an element in the vector.

Reply:
Okay, that may work due to how operator== is implemented, but removing while iterating over a vector with range-for loop is still a bad idea. After things are moved inside the vector due to removeOne() you are going to skip an element.

Reply:
Yeah. I'll refactor that.